File: go/word-count/word_count.go
// Word Count: Separate a phrase into words and count how many times a word occurs in the phrase.
package wordcount

import (
	"regexp"
	"strings"
)

type Frequency map[string]int

// Count the occurences of words in a string.
// @param string: The text to extract words from
// @returns: a Frequency type (map[string]int) that has keys of words (lower case) with a value of their counts.
// will return an empty set if no words are found or the regexp used doesn't compile.
func WordCount(phrase string) Frequency {
	ret := Frequency{}
	re, err := regexp.Compile(`\w+('\w+)?`)
	if err != nil {
		return ret // empty on error.
	}
	results := re.FindAllString(phrase, -1)
	for _, word := range results {
		ret[strings.ToLower(word)]++
	}

	return ret
}

File: php/word-count/WordCount.php
<?php
// Word Count exercise.

declare(strict_types=1);

// Return a hash map (array) of the words in the string to the number of occurences.
// @param $words: The string to pattern match on.
// @returns: Array with word as key (in lower case) and number of ocurrences as the value.
function wordCount(string $words): array
{
    $pattern = "/[a-z0-9]+(\'[a-z]+)?/im";
    preg_match_all($pattern, strtolower($words), $result);
    return array_count_values($result[0]);
}
